### PR TITLE
Add histogram plot for posterior predictives

### PR DESCRIPTION
**Pull request by fanshi118**
**Thank your for opening a PR!**

Here are a few important guidelines and requirements to check before your PR can be merged:
+ [x] There is an informative high-level description of the changes.
+ [x] The description and/or commit message(s) references the relevant GitHub issue(s).
+ [x] The commit messages follow [these guidelines](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
+ [x] The commits correspond to [relevant logical changes](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes).
+ [x] There are tests covering the changes introduced in the PR.

Don't worry, your PR doesn't need to be in perfect order to submit it.  As development progresses and/or reviewers request changes, you can always [rewrite the history](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#_rewriting_history) of your feature/PR branches.

If your PR is an ongoing effort and you would like to involve us in the process, simply make it a [draft PR](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/about-pull-requests#draft-pull-requests).

### CODE REVIEW

**Review by brandonwillard — CHANGES_REQUESTED** [review]
Here are the changes we discussed, plus one extra:
- [ ] Load `pickle`d results from the zero/Poisson and time-varying examples and use those to demonstrate the plot results (without the old example code)
- [ ] Put the plot-producing code into a single function.  That will simplify the timing mentioned below.
- [ ] Try using `np.histogram(..., normed=True)` instead of `dens = freqs / np.max(freqs)`.
- [ ] Use the `%%timeit -n 1 -r 0` magic (or something like that) to time a single invocation of the plotting code, because we want to know how long it takes

File: pymc3_hmm/utils.py [hunk 1]
```diff
@@ -362,3 +363,69 @@ def plot_fn(ax, data, **kwargs):
...
+    axes.set_xlim(xmin=x_dt[0], xmax=x_dt[n_points])
+    axes.set_title(
+        "Posterior Preditives (color-coded by normalized bin weights)", fontsize=15
+    )
+    axes.tick_params(labelsize=11.5)
+    axes.set_xlabel("DateTime Hour", fontsize=14)
+    axes.set_ylabel("Posterior Predictives Histogram", fontsize=14)
+    legend_lines, legend_labels = axes.get_legend_handles_labels()
+    axes.legend(legend_lines, legend_labels)
+    plt.tight_layout()
```
Comment (brandonwillard) [inline pymc3_hmm/utils.py:431]: This function is only needed for your local testing.

**Review by brandonwillard — CHANGES_REQUESTED** [review]

**Comment by brandonwillard** [comment]
We can ignore the coverage requirements for that new plotting function (i.e. set `# pragma: no cover` on the function).

**Comment by fanshi118** [comment]
> We can ignore the coverage requirements for that new plotting function (i.e. set `# pragma: no cover` on the function).

Just made the update: got rid of the wrapper and rearranged the commits.

**Comment by fanshi118** [comment]
@brandonwillard - just rebased. Moved the wrapper/styling code from `utils.py` to the notebook, returning `axes` in `plot_predictive_histograms`.

**Review by brandonwillard — CHANGES_REQUESTED** [review]
I don't think we need to add the demo notebook when we merge this.  It would be good to make it available independently as a Gist, though.

**Comment by fanshi118** [comment]
> I don't think we need to add the demo notebook when we merge this. It would be good to make it available independently as a Gist, though.

Removed; Gist addd [here](https://gist.github.com/fanshi118/ca32892877c2b2d75323cbff014ab800).

**Review by brandonwillard — CHANGES_REQUESTED** [review]
I've made some updates to the Poisson/zero mixture example notebook.  Take a look at the out-of-sample posterior predictive plot produced by `plot_predictive_histograms`; it adds a `fill_between` for the values between the two mixture components, but there's effectively zero weight in the corresponding bins.

Let's see if we can remove the `fill_between` coloring for bins with zero weight.

**Review by brandonwillard — CHANGES_REQUESTED** [review]
This looks great!

Can you make the same plotting and out-of-sample updates to the other example notebook?  That example should produce a much more interesting out-of-sample plot; one for which this new plotting function should really shine.

**Comment by fanshi118** [comment]
> Can you make the same plotting and out-of-sample updates to the other example notebook? That example should produce a much more interesting out-of-sample plot; one for which this new plotting function should really shine.

The time-varying model has time-dependent variables, which require the shape parameters to be reset as we sample the out-of-sample posterior predictives. This likely will lead to using the design matrix, for us to be able to update the time-dependent variables as we'd like to in OOS. I updated the notebook so that it drops the `S_t` and `P_t`, assumes the same shape, and truncates the series to the first week, just to show how the histogram plot would look. @brandonwillard , let me know what you think of this as opposed to potentially using the design matrix for this example.

**Comment by brandonwillard** [comment]
Have you considered using [`pcolorfast`](https://matplotlib.org/3.1.0/api/_as_gen/matplotlib.axes.Axes.pcolorfast.html#matplotlib-axes-axes-pcolorfast) instead of `fill_between`?  I'm not sure if it would make for a good alternative, but it seems like it.

**Pull request merged**